### DISCUSSION TITLE
Confusing duplicate items in POI overlay chooser

### DISCUSSION THREAD

**Discussion by snouup** [osmandapp/OsmAnd#12234] [Ideas]
### Description

There are a few ways to get a POI type listed in the POI overlay chooser:

1. Inspecting a POI on the map temporarily adds it's POI type to the list, only until you restart osmand
2. Adding POI type to list of search categories (via Search -> Save filter as) adds it permanently
3. Sometimes, just doing a search will add one temporarily I think. Not quiet sure how/when

Often after adding a new poi type to categories, I see a duplicate entry in the poi chooser, which at first glance looks like a bug, especially when the duplicate disappears on a restart.

Possible improvements
- hide duplicate
- differentiate implicitly added poi types somehow, communicate what's going on - maybe a separate section titled "recent" containing the X most recent items?
- or don't implicitly add pois at all? (Did anyone ask for it? I personally could do without...)

### How to reproduce?

Inspect some poi
Search for its type, save filter
Check duplicate in poi overlay list

### Your Environment
OsmAnd Version: OsmAnd~ 3.9.10

**Comment by sonora** [comment]
Side note: Also here, just like for tracks, we have the issue that the POI Overlay dialog overlaps in functionality with our search screen, thus duplicating the set of possible bugs and behavioural issues.

Before fixing either one, maybe it is worth to try to unify to just one dialog, throw the other one out, then start fixing issues?

**Comment by snouup** [comment]
Oof, I don't know. The Search is already one of the most complex dialogs in osmand, do we want to make it more complicated?
As for me, I'm grateful for the quick and clean access to poi overlay from profile settings. Sometimes I'll switch some poi on/off on the fly, much quicker than maintaining yet another profile.

**Comment by sonora** [comment]
The Search screen could simply sport an additional tab of things currently displayed, or perhaps we can even put that on the existing Categories tab.

Even today, from the POI Overlay's screen we already link to the Search's Category tab, which shows how closely related this is.

There may be little logical justification to keep the POI Overlay as a separate screen. Plus, we are a little inconsistent, as you note in this issue, showcasing in one place which items and serach results a user currently picked to be displayed, and without duplicates.

**Reply by Lee-Carre** [reply]
> There may be little logical justification to keep the POI Overlay as a separate screen.

It's also used for the `Show along the route` function (of\|during) navigation.

**Comment by snouup** [comment]
I'm warming up to it. Allow me some counterpoints still:

- Profile quick section mainly affects the display of map (map style, shown details, etc.) Poi overlay fits into this as a substantial part of display (assuming you show enough poi's). For me that section is all about changing what I _see_
- Search screen can be more of a customisation tool that is used beforehand (before a trip), making custom filters for later use, or search for specific poi to bookmark it. I think of that screen as allowing me to change my _toolbox_ that I will later use. Yes, there are a few operations breaking this model, but overall it fits.

I kind of like that two step approach, separating preparation from day-to-day use. The current UI is well suited to this, showing you a large, fullscreen dialog for Search (powerful toggles, etc) where you can tweak your toolbox in a map-agnostic way at first, then, during day to day use, you can do tweaking side by side with the map, and see the effect of changes instantly.

Maybe related: To what extent do we associate categories and overlays with a specific profile? I personally have overlays tied to profiles but my categories are the same for all profiles. Did I end up with this setup for any logical reason, or did the current UX nudge me towards it? Time will tell.

**Comment by vshcherb** [comment]
I'm still not sure what kind of duplicate it is? 
POI Overlay dialog is exactly the same as Search screen and that how it was and should be, so there is no changes planned here.

**Reply by scaidermern** [reply]
> I'm still not sure what kind of duplicate it is?

The temporary POI categories can be created the following way:

- Enter search
- Search for a specific POI category, let's say "museum"
- Tap on "museum" in the result list
- Tap on the filter button in the top right corner
- Select anything there, let's say "opened now"
- Tap on apply
- Close the result list
- Now open Configure map -> POI Overlay
- There will be a new "museum" category

This "temporary" category will exist until you close and restart OsmAnd. If you already created a similar custom POI category before, this will lead to a duplicate. 

Also this "temporary" category won't appear under Search > Categories. So you can't modify/remove it manually. I think this category should get deleted automatically as soon as you close the search menu.

**Reply by vshcherb** [reply]
Yes, that's right. Though I don't see any issue here, it's convenient that it stays for a while

**Reply by scaidermern** [reply]
> Yes, that's right. Though I don't see any issue here, it's convenient that it stays for a while

Is it?

Shouldn't it stay then in both the POI overlay menu _and_ in the search menu? It gets created in the search menu, then disappears there and reappears in the POI overlay menu. This is very confusing and I don't see the benefit of it. If the user wants it to stay for a while, he can save the filter.

**Reply by vshcherb** [reply]
Search menu is not the menu of displayable filters, so it's permanent and doesn't change. Indeed from Configure Map -> Poi Overlay you can't save and edit filter further and you can't get back to search results. That could be improved .

**Reply by Lee-Carre** [reply]
> There may be little logical justification to keep the POI Overlay as a separate screen.

That would need to be clearly communicated to the user, through the UI (and I'm not thinking of a textual warning message), lest the golden rule (of treating user-data as sacred) is violated.

Short of that, better for it to persist and require manual deletion, than to disappear (at least, not before the user is unlikely to need it, again; which seems closely related to the current session life, like vshcherb describes).
